### PR TITLE
`Development`: Add test exam management e2e tests

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Currently there are no E2E tests that test the creation of groups within a tetst exam and the edit of groups.

### Description

With this PR we add new test exam managment E2E tests. We add a exercise group and a text exercise. 

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test

### CODE REVIEW

**Review by julian-christl — COMMENTED** [review]
There are no changes to the production code, so no testing is required.

Regarding the tests, I have a question: It looks to me like one test is based on the next one.
I think that's bad practice as each test case should be isolated.
In other Cypress tests that create and delete entities I have previously seen a creation before each and deletion after each.

What do you think?

**Comment by TheZoker** [comment]
@julian-christl Thanks for the hint and you are totally right! I refactored the new test exam tests and also the existing exam managment tests to follow this rule. All tests should now be independent of each other

**Review by julian-christl — DISMISSED** [review]
Thanks for the changes, looks good to me now. E2E tests are succeeding.

File: src/test/cypress/e2e/exam/ExamManagement.cy.ts [hunk 1]
```diff
@@ -37,128 +41,145 @@ describe('Exam management', () => {
...
+            cy.visit('/');
+            navigationBar.openCourseManagement();
+            courseManagement.openExamsOfCourse(course.shortName!);
+            examManagement.openExerciseGroups(exam.id!);
+            // If the group in the "Create group test" was created successfully, we delete it so there is no group with no exercise
+            if (createdGroup) {
+                exerciseGroups.clickDeleteGroup(createdGroup.id!, createdGroup.title!);
+            } else {
+                exerciseGroups.clickDeleteGroup(exerciseGroup.id!, exerciseGroup.title!);
+            }
```
Comment (rriyaldhi) [inline src/test/cypress/e2e/exam/ExamManagement.cy.ts:158]: Should there be an assertion which verifies that the group has been deleted?

File: src/test/cypress/e2e/exam/test-exam/TestExamManagement.cy.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            navigationBar.openCourseManagement();
+            courseManagement.openExamsOfCourse(course.shortName!);
+            examManagement.openExerciseGroups(exam.id!);
+            // If the group in the "Create group test" was created successfully, we delete it so there is no group with no exercise
+            if (createdGroup) {
+                exerciseGroups.clickDeleteGroup(createdGroup.id!, createdGroup.title!);
+            } else {
+                exerciseGroups.clickDeleteGroup(exerciseGroup.id!, exerciseGroup.title!);
+            }
+        });
```
Comment (rriyaldhi) [inline src/test/cypress/e2e/exam/test-exam/TestExamManagement.cy.ts:157]: Should there be an assertion which verifies that the group has been deleted?

**Review by rriyaldhi — COMMENTED** [review]
Looks good! Just left 2 small questions.

**Review by 4ludwig4 — APPROVED** [review]
Checked E2E pipeline and also took a look at the code. LGTM

**Review by akesfeden — APPROVED** [review]
E2E tests pass on Bamboo and the changes seem reasonable.

File: src/test/cypress/e2e/exam/ExamManagement.cy.ts (hunk 1 above)
Comment (TheZoker) [reply src/test/cypress/e2e/exam/ExamManagement.cy.ts:157]: Thanks for the hint, I implemented it in the latest commit 🎉

File: src/test/cypress/e2e/exam/test-exam/TestExamManagement.cy.ts (hunk 2 above)
Comment (TheZoker) [reply src/test/cypress/e2e/exam/test-exam/TestExamManagement.cy.ts:158]: Thanks for the hint, I implemented it in the latest commit 🎉

**Review by akesfeden — APPROVED** [review]
Changes since my last approval look good.

**Review by rriyaldhi — APPROVED** [review]
Thanks for applying the suggestion.

**Review by 4ludwig4 — APPROVED** [review]

**Pull request merged**